### PR TITLE
(SIMP-MAINT) Rakefile split and dockerfile fix

### PR DESCRIPTION
**Pull request by trevor-vaughan**
* Split the rakefile into more maintainable parts
* Updated the EL8 dockerfile to work with minimum-version EL8 builds
  needed for SELinux

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1,155 +1,394 @@
...
+  <<: *setup_bundler_env
+  script:
+    - 'bundle exec rake metadata_lint'
 
-# Anchor to disable release flavored tests, and run them only when
-# the environment variable SIMP_RELEASE_TESTS is set in the GitLab repo settings
-.only_with_SIMP_RELEASE_TESTS: &only_with_SIMP_RELEASE_TESTS
-  only:
-    variables:
-      - $SIMP_RELEASE_TESTS
```
Comment (lnemsick-simp) [inline .gitlab-ci.yml:44]: The variables specific to this project need to be reconfigured

**Review by lnemsick-simp — CHANGES_REQUESTED** [review]

File: build/Dockerfiles/SIMP_EL8_Build.dockerfile [hunk 2]
```diff
@@ -80,8 +97,8 @@ RUN runuser build_user -l -c "for i in {1..5}; do { gpg2 --keyserver  hkp://pool
 RUN runuser build_user -l -c "for i in {1..5}; do { gpg2 --keyserver  hkp://pool.sks-keyservers.net --recv-keys 7D2BAF1CF37B13E2069D6956105BD0E739499BDB || gpg2 --keyserver hkp://pgp.mit.edu --recv-keys 7D2BAF1CF37B13E2069D6956105BD0E739499BDB || gpg2 --keyserver hkp://keys.gnupg.net --recv-keys 7D2BAF1CF37B13E2069D6956105BD0E739499BDB; } && break || sleep 1; done"
 #RUN runuser build_user -l -c "gpg2 --refresh-keys"
 RUN runuser build_user -l -c "curl -sSL https://raw.githubusercontent.com/rvm/rvm/stable/binscripts/rvm-installer -o rvm-installer && curl -sSL https://raw.githubusercontent.com/rvm/rvm/stable/binscripts/rvm-installer.asc -o rvm-installer.asc && gpg2 --verify rvm-installer.asc rvm-installer && bash rvm-installer"
-RUN runuser build_user -l -c "rvm install 2.4.10 --disable-binary"
-RUN runuser build_user -l -c "rvm use --default 2.4.10"
+RUN runuser build_user -l -c "rvm install 2.6"
```
Comment (lnemsick-simp) [inline build/Dockerfiles/SIMP_EL8_Build.dockerfile:100]: Why Ruby 2.6?

Comment (trevor-vaughan) [reply build/Dockerfiles/SIMP_EL8_Build.dockerfile:100]: Works with the least amount of fuss. 2.4 and 2.5 have some quirks and 2.7 is 3.0 prep.

**Comment by trevor-vaughan** [comment]
Marked as WIP because I need to fix the EL7 dockerfile (again)

**Review by lnemsick-simp — COMMENTED** [review]
Have verified the Rakefile changes work and we can build an EL7 ISO with build:auto.  More evaluation to follow.

File: .github/workflows/pr_tests.yml [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: 'Install Ruby ${{matrix.puppet.ruby_version}}'
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.5
+          bundler-cache: true
+      - name: 'Tags and changelogs'
+        run: |
+          bundle exec rake pkg:check_version
```
Comment (lnemsick-simp) [inline .github/workflows/pr_tests.yml:101]: These three pkg rake tasks to not exist for simp-core and so this check will fail.

File: .github/workflows/pr_tests.yml [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        run: 'bundle exec pdk build --force'
+
+  spec-tests:
+    name: 'Puppet Spec'
+    needs: [puppet-syntax]
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        puppet:
+          - label: 'Puppet 6.18 [SIMP 6.5/PE 2019.8]'
```
Comment (lnemsick-simp) [inline .github/workflows/pr_tests.yml:114]: These permutations may not be needed for the limited unit tests in this project.

**Comment by lnemsick-simp** [comment]
Have verified the Docker files work and are largely usable for simp-core builds. (EL8 is not ready for ISO building yet, but can build all but the pupmod-simp-vox_selinux RPM).

**Review by lnemsick-simp — APPROVED** [review]

**Comment by lnemsick-simp** [comment]
https://gitlab.com/simp/simp-core/-/pipelines/251685509

**Comment by trevor-vaughan** [comment]
Failure issue is outside of this module. Will push and we'll get the LDAP module updated.

**Pull request merged**